src/benchmarks/micro/libraries/System.IO/StreamReaderReadToEndTests.cs

```diff
@@ -0,0 +1,34 @@
+using System.Text;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.IO.Tests;
+
+[BenchmarkCategory(Categories.Libraries)]
+public class StreamReaderReadToEndTests : TextReaderReadLineTests
+{
+    private byte[] _bytes;
+
+    [GlobalSetup]
+    public void GlobalSetup()
+    {
+        _text = GenerateLinesText(LineLengthRange, 16 * 1024);
+        _bytes = Encoding.UTF8.GetBytes(_text);
+    }
+
+    [Benchmark]
+    public void ReadToEnd()
+    {
+        using StreamReader reader = new (new MemoryStream(_bytes));
```

Comment:
Should the MemoryStream be constructed in `GlobalSetup` instead?
In the benchmark method the `Position` should be set to `0` for the MS.

So to focus solely on `ReadToEnd`, and not constructing additional objects.
In that vein the `StreamReader` could also be constructed outside the benchmark-method?

Reply:
That makes sense. Will do that.

Reply:
Done.

Reply:
@gfoidl Have a look at it again when you can.